### PR TITLE
Use random.choices when available for better performance

### PR DESCRIPTION
**Pull request by catleeball**

### CODE REVIEW

**Comment by catleeball** [comment]
PR for bug #421

File: faker/utils/distribution.py [hunk 1]
```diff
@@ -17,10 +18,13 @@ def cumsum(it):
...
     assert len(a) == len(p)
 
-    cdf = list(cumsum(p))
-    normal = cdf[-1]
-    cdf2 = [float(i) / float(normal) for i in cdf]
-    uniform_sample = random_sample()
-    idx = bisect.bisect_right(cdf2, uniform_sample)
-
-    return a[idx]
+    if version_info.major >= 3 and version_info.minor >= 6:
```
Comment (fcurella) [inline faker/utils/distribution.py]: is checking against `hasattr(random, 'choices') any slower?

Comment (catleeball) [inline faker/utils/distribution.py]: Good call! Looks like hasattr is quite a bit faster, at least when testing on python 3.6 on my machine.

```
$ python3 -m timeit "import random; hasattr(random, 'choices')"
1000000 loops, best of 3: 0.265 usec per loop
$ python3 -m timeit "import sys; sys.version_info.major >= 3 and sys.version_info.minor >= 6"
1000000 loops, best of 3: 0.614 usec per loop
```

I'll update the PR soon. :)

**Comment by catleeball** [comment]
Okay, updated with `hasattr()`. Unit tests seem happy. Let me know if any further edits are needed. :)

**Comment by jchildren** [comment]
How many times did you run the unit tests? Success seemed to be randomized. Is it worth adding new ones that are more thorough?

File: faker/utils/distribution.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # coding=utf-8
 
 import bisect
+from sys import version_info
```
Comment (fcurella) [inline faker/utils/distribution.py]: You can just remove this import, since it's not used anymore

Comment (catleeball) [inline faker/utils/distribution.py]: Oh, good catch, I'll remove that!

**Comment by catleeball** [comment]
Okay, just pushed a commit removing the unused import. :)

@jchildren: Good catch! To be sure, I ran the tests 100 times with `for i in {1..100}; do python3 setup.py test &> ~/100tests.log; done` and got all passing output: https://drive.google.com/file/d/0B4p5S5l8DBDxcWxnR3JfUy1PcDQ/view?usp=sharing

**Comment by fcurella** [comment]
perfect! Thank you so much for working on this, it's deeply appreciated! 🚀 ✨

**Pull request merged**